k9mail/src/main/java/com/fsck/k9/crypto/MessageDecryptVerifier.java

```diff
@@ -200,11 +200,9 @@ private static Part findPrimaryPartInAlternative(Part part) {
             if (body instanceof Multipart) {
                 Multipart multi = (Multipart) body;
                 BodyPart signatureBody = multi.getBodyPart(1);
-                if (isSameMimeType(signatureBody.getMimeType(), APPLICATION_PGP_SIGNATURE)) {
-                    ByteArrayOutputStream bos = new ByteArrayOutputStream();
-                    signatureBody.getBody().writeTo(bos);
-                    return bos.toByteArray();
-                }
+                ByteArrayOutputStream bos = new ByteArrayOutputStream();
```

Comment:
Why did you remove the MIME type check?

Please create a separate commit for such changes. That way you can write down what motivated this.

Reply:
Sorry, separate commit would have been the way to go. This check should happen before a part is even considered for parsing as pgp/mime signed, if it doesn't match here it means we screwed up before